library/std/src/collections/hash/set.rs

```diff
@@ -65,8 +65,8 @@ use super::map::{map_try_reserve_error, RandomState};
 /// ```
 ///
 /// The easiest way to use `HashSet` with a custom type is to derive
-/// [`Eq`] and [`Hash`]. We must also derive [`PartialEq`], this will in the
-/// future be implied by [`Eq`].
+/// [`Eq`] and [`Hash`]. We must also derive [`PartialEq`],
+/// which is implied by [`Eq`].
 ///
```

Comment:
i don't think this change is correct. the note is saying "we have to derive PartialEq because Eq without PartialEq will error, and at some point we want to change derive(Eq) to imply derive(PartialEq)". that second half still isn't the case today — it's probably true that it's never going to happen, but i would expect the new comment to look a little different if that's what you meant.

Reply:
I suspected that could be the case, but got confused by:
> This property cannot be checked by the compiler, and therefore Eq implies [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html), and has no extra methods.

from https://doc.rust-lang.org/std/cmp/trait.Eq.html

I guess _implies_ means something else in that example?

Reply:
yes, unfortunately. the one here is using it to describe a *behavior* of the compiler and the one on the trait page is using it to describe a *property* of the language (that your code won't compile if you have Eq without PartialEq).

Reply:
please see if latest wording is acceptable